pkgs/top-level/all-packages.nix

```diff
@@ -34021,10 +34021,18 @@ with pkgs;
 
   xdotool = callPackage ../tools/X11/xdotool { };
 
-  xenPackages = recurseIntoAttrs (callPackage ../applications/virtualization/xen/packages.nix {});
-
-  xen = xenPackages.xen_4_19;
-  xen-slim = xenPackages.xen_4_19-slim;
+  buildXenPackage = callPackage ../build-support/xen { };
+
+  inherit (callPackages ../by-name/xe/xen { })
```

Comment:
You shouldn't directly import by-name packages

inherit (xen) xen_4_19 ...;

Might work, but I don't think we want that.
As I can see, by-name is somehow supposed to work with pkgsets, but I can't figure out how.

Reply:
That's a straight up copy of the documentation. https://github.com/NixOS/nixpkgs/blob/8637e879e425363b14aa23f3ec1f69e32c9f512f/pkgs/by-name/README.md?plain=1#L183-L201

Reply:
They don't import from by-name, all by-name packages are automatically discovered and imported

Reply:
Except if the package.nix refers to multiple versions, in which case, `callPackages` in `all-packages.nix` is required.

Reply:
I think I understand what this documentation wants us to do, I'll try

Reply:
Yep, It works for me:

by-name.patch

Reply:
How did I do that, lol

by-name.patch

Reply:
Github is acting weird:
https://gist.github.com/CertainLach/5fe65ffe618ee4cab917463b496a39a0

Reply:
maybe send on the xen room if GH won't let you?

Reply:
all in favour of just putting `ocamlPackages_4_14` in the generic builder and not looking back? it hasn't changed in years

Reply:
Not familiar with ocaml packaging in NixOS, can't we just use last version, or some breaking change breaks xen?

Reply:
the 4.19 build looks fine. if `nixpkgs-review` passes I'm inclined to just send it.

Reply:
I'll finish my rework of this PR and start nixpkgs-review.

Reply:
NVM, it is still invalid.
    For each package directory, pkgs.lib.isDerivation pkgs.${name} must be true.

Reply:
Waiting for nixpkgs-review on https://github.com/CertainLach/nixpkgs/commit/183eae6d051258b85257d45dfde415266b607451